openmls/src/group/mls_group/creation.rs

```diff
@@ -189,3 +189,100 @@ impl MlsGroup {
         ))
     }
 }
+
+impl StagedMlsGroup {
+    /// Creates a new staged group from a [`Welcome`] message. Returns an error
+    /// ([`WelcomeError::NoMatchingKeyPackage`]) if no [`KeyPackage`]
+    /// can be found.
+    /// Note: calling this function will consume the key material for decrypting the [`Welcome`]
+    /// message, even if the caller does not turn the [`StagedMlsGroup`] into an [`MlsGroup`].
+    // TODO: #1326 This should take an MlsMessage rather than a Welcome message.
```

Comment:
Should we do that here? Since it's a new funtion we can make it better from the start 🙂